clients/client-rds-data/RdsDataServiceClient.ts

```diff
@@ -73,12 +73,25 @@ export type ServiceOutputTypes =
   | ExecuteStatementResponse
   | BatchExecuteStatementResponse;
 
-export interface RDSDataRuntimeDependencies {
+export interface RDSDataSharedConfig {
   /**
    * The function that will be used to populate serializing protocol
    */
-  protocol: string;
-
+  protocol?: string;
+  /**
+   * The service name with which to sign requests.
+   */
+  signingName?: string;
```

Comment:
Can this be moved to AwsAuthInputConfig? If we do that, then this interface is left with just protocol and apiVersion, both of which could just be added as required configuration of a SmithyClient. Then this interface could go away.